### PR TITLE
Complete models and test methods

### PR DESCRIPTION
**Pull request by David-Lanzz**
# HELLO REVIEWER!!!

## I have completed the tasks assigned to me, please review them and get back to me.
- No linter errors
- Correct use of git-flow
- Creation of models and foreign keys
- Task completion
## Any issues or suggestions, please let me know via the comment section of this PR

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+
```
Comment (rustam-tolipov) [inline app/models/user.rb:5]: I can see you put a greate efford on this project!
However, there are placed that need to be fixed before approval! Let's get started. 

- [ ] Your user has many posts also has many comments and likes as you declared posts

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Like < ApplicationRecord
+  belongs_to :post
```
Comment (rustam-tolipov) [inline app/models/like.rb:2]: - [ ] Your like is belongs to user and post equally. Kindly update it.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Comment < ApplicationRecord
+  belongs_to :post
```
Comment (rustam-tolipov) [inline app/models/comment.rb:2]: - [ ] As likes your comments belongs to user and post.

File: app/models/post.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
```
Comment (rustam-tolipov) [inline app/models/post.rb:3]: - [ ] Add connection between post and likes, remember post has many likes.

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+  def update_post_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(3)
```
Comment (rustam-tolipov) [inline app/models/post.rb]: - [ ] You required to display five recent comments, not three. Kindly update it!

File: app/models/like.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+  def update_likes_counter
```
Comment (rustam-tolipov) [inline app/models/like.rb:7]: - [ ] Also both update..counter methods should be called before or after creating. By just creating the model wont help you to update the table. More information: [before_save...](https://guides.rubyonrails.org/active_record_callbacks.html)

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @David-Lanzz,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Good readme_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 4 above)
Comment (brytebee) [reply app/models/post.rb:3]: Kindly update this model as suggested by the previous reviewer, a post can have multiple likes.

Consider this code below:

```ruby
has_many :likes
```

File: app/models/post.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+
+  def update_post_counter
```
Comment (brytebee) [inline app/models/post.rb:8]: This method is never called as the previous reviewer pointed out.
Ensure you call it before or after save.

Consider the code snippet bellow.

```ruby
after_save: update_post_counter
```

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @David-Lanzz 👋, 

##

### Highlights💡
- Good DB practice and data model understanding.✔️ 

##

Great work on making the changes 👏 👏 
You've done well implementing some of the requested changes requested by the previous review, but there are still some which aren't addressed yet.


# Suggested changes
_Check the comments under the review._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ 🎉 ✔️ 

Hi @David-Lanzz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] All requested changes are implemented ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] No linters errors ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can get the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**